python/nano/test/test_models_lightning_support.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from unittest import TestCase
+
+import torch
+from test._train_torch_lightning import train_torch_lightning
+from torch import nn
+
+from bigdl.nano.pytorch.vision.models import lightning_support, vision
+
+config = {
+    'lr': 0.01,
+    'optim': 'Adam',
+}
+batch_size = 256
+num_workers = 0
+data_dir = os.path.join(os.path.dirname(__file__), "data")
+
+
+def loss_creator(config):
+    return nn.CrossEntropyLoss()
+
+
+def optimizer_creator(model, config):
+    return getattr(torch.optim, config.get("optim", "Adam"))(model.parameters(), lr=config.get("lr", 0.001))
+
+
+@lightning_support.lightning_module(loss_creator, optimizer_creator, config)
```

Comment:
**It seems to me that the `trainer.compile` has a better user experience** since you can not create a opt instance before the model instance is built, which leads to some config dict. 

Still both of the methods involve a abnormal usage (i.e. multiple-decorator and `trainer.compile`). We need to give thorough and detailed user guide and in-code-warning.

Reply:
> **It seems to me that the `trainer.compile` has a better user experience** since you can not create a opt instance before the model instance is built, which leads to some config dict in the decorator case.
> 
> Still the method involve a abnormal usage (i.e. `trainer.compile`). We need to give thorough and detailed user guide and in-code-warning.
> 
> btw, we don't need to implement `onnx=True` parameter in this PR since I have not merged the onnx PR. I will do it later after this PR has been merged.

What do you mean by abnormal usage? I think the point is
1) If the user simply has a PyTorch model, he or she can directly use it in `nano.pytorch.trainer` methods (if the default behavior of `fit`, `test`, `predict`, etc., works for him or her)

2) If the user needs more complex behavior (e.g., onnxruntime support), he or she needs to explictly convert it to pl_module, and we can provide an API based on either decorator or compile method.

Reply:
> > **It seems to me that the `trainer.compile` has a better user experience** since you can not create a opt instance before the model instance is built, which leads to some config dict in the decorator case.
> > Still the method involve a abnormal usage (i.e. `trainer.compile`). We need to give thorough and detailed user guide and in-code-warning.
> > btw, we don't need to implement `onnx=True` parameter in this PR since I have not merged the onnx PR. I will do it later after this PR has been merged.
> 
> What do you mean by abnormal usage? I think the point is
> 
> 1. If the user simply has a PyTorch model, he or she can directly use it in `nano.pytorch.trainer` methods (if the default behavior of `fit`, `test`, `predict`, etc., works for him or her)
> 2. If the user needs more complex behavior (e.g., onnxruntime support), he or she needs to explictly convert it to pl_module, and we can provide an API based on either decorator or compile method.

I think we can just ask all users with a nn.module to do this:
```python
model = Net()
model_pl = trainer.compile(model, loss, opt, onnx=bool)
# then use model_pl (pl.lightningmodule) to do anything else
```
abnormal usage means that `trainer.compile` is not a easy name for users to come up with directly, we need examples, quickstarts and user guides to guide them.

Reply:
It does not have to be `trainer.compile`; maybe something similar to `ray.distribtued`, such as `nano.pl_module`